packages/node/src/starknet/api.starknet.ts

```diff
@@ -80,6 +76,7 @@ export class StarknetApi implements ApiWrapper {
   private contractInterfaces: Record<string, Abi> = {};
   private chainId?: string;
   private specVersion?: string;
+  private finalizedBlockService?: FinalizedBlockService;
```

Comment:
I don't think this should be optional. It would always exist.

Maybe its a typescript thing but you could un-nest the logic by doing this:
```ts
if (protocolStr !== 'https' && protocolStr !== 'http') {
    throw new Error(`Unsupported protocol: ${protocol}`);
}
```